internal/k8shandler/reconciler.go

```diff
@@ -67,37 +80,31 @@ func Reconcile(requestCluster *logging.ClusterLogging, requestClient client.Clie
 
 	// Reconcile Collection
 	if err = clusterLoggingRequest.CreateOrUpdateCollection(); err != nil {
+		telemetry.Data.CLInfo.Set("healthStatus", constants.UnHealthyStatus)
 		telemetry.Data.CollectorErrorCount.Inc("CollectorErrorCount")
+		telemetry.UpdateCLMetricsNoErr()
 		return fmt.Errorf("unable to create or update collection for %q: %v", clusterLoggingRequest.Cluster.Name, err)
 	}
 
 	// Reconcile metrics Dashboards
 	if err = metrics.ReconcileDashboards(clusterLoggingRequest.Client, reader); err != nil {
+		telemetry.Data.CLInfo.Set("healthStatus", constants.UnHealthyStatus)
+		telemetry.UpdateCLMetricsNoErr()
 		log.Error(err, "Unable to create or update metrics dashboards", "clusterName", clusterLoggingRequest.Cluster.Name)
```

Comment:
This doesn't return an error, will fall thru and overwrite with HealthyStatus. Suggest setting HealthyStatus at the top of this function, and it will get over-written with UnhealthyStatus if an error is encountered.

Reply:
@jcantrill Is it correct that no error is returned here? If so is this really an unhealthy condition?